src/sdk/pynni/nni/smac_tuner/smac_tuner.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge,
+# to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+'''
+smac_tuner.py
+'''
+
+from nni.tuner import Tuner
+
+import sys
+import logging
+import numpy as np
+import json_tricks
+from enum import Enum, unique
+from .convert_ss_to_scenario import generate_scenario
+
+from smac.utils.io.cmd_reader import CMDReader
+from smac.scenario.scenario import Scenario
+from smac.facade.smac_facade import SMAC
+from smac.facade.roar_facade import ROAR
+from smac.facade.epils_facade import EPILS
+
+@unique
+class OptimizeMode(Enum):
+    '''
+    Oprimize Mode class
+    '''
+    Minimize = 'minimize'
+    Maximize = 'maximize'
+
+class SMACTuner(Tuner):
+    def __init__(self, optimize_mode):
+        '''
+        Constructor
+        '''
+        self.logger = logging.getLogger(
```

Comment:
I'am sure the logger instance is Zheliu's style?

Reply:
not the style we use, smac uses its own logging, it would be better to just keep it.